### PR TITLE
Enable Megatron-LM workload on ROCm

### PR DESCRIPTION
**Pull request by rraminen**
This PR contains the changes to

* Enable Megatron workload on ROCm
* Add extra_include_paths to hipify header files
* Fix to resolve the rsqrtf() call to device function from host function error on ROCm
*  auto -> int conversion on ROCm as a workaround for hipify error

### CODE REVIEW

**Comment by rraminen** [comment]
@jeffra, could you please review this PR?

**Review by jithunnair-amd — APPROVED** [review]

File: megatron/fused_kernels/__init__.py [hunk 1]
```diff
@@ -31,62 +32,87 @@ def load(args):
...
-            sources=sources,
-            build_directory=buildpath,
-            extra_cflags=['-O3',],
-            extra_cuda_cflags=['-O3',
-                               '-gencode', 'arch=compute_70,code=sm_70',
-                               '--use_fast_math'] + extra_cuda_flags + cc_flag,
-            verbose=(args.rank == 0)
-        )
+    def _cpp_extention_load_helper(name, sources, extra_cuda_flags, extra_include_paths):
+        if torch.version.hip is not None:
```
Comment (tjruwase) [inline megatron/fused_kernels/__init__.py:49]: I feel this code could be simplified so that the only branch is on the value of `extra_cuda_cflags`.

Comment (rraminen) [reply megatron/fused_kernels/__init__.py:49]: I made the suggested change.

cc: @jithunnair-amd

Comment (tjruwase) [reply megatron/fused_kernels/__init__.py:49]: @rraminen, I apologize for my vague feedback. This is not what I meant. 

What I meant was that the 2 return commands only seem to differ in the value of `extra_cuda_cflags`. So, I suggested having just one return like
```python
    if torch.version.hip is not None:
       extra_cuda_cflags = ...
    else:
       extra_cuda_cflags = ...

    return cpp_extension.load(
                ...
      )

Comment (rraminen) [reply megatron/fused_kernels/__init__.py:49]: Sorry, I made the changes.

**Comment by jithunnair-amd** [comment]
@jeffra @tjruwase Please let us know if you have any other comments.

**Review by tjruwase — APPROVED** [review]

**Comment by tjruwase** [comment]
> @jeffra @tjruwase Please let us know if you have any other comments.

Apologies for the delay. Looks good to me.

**Review by jeffra — APPROVED** [review]

**Pull request merged**